pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -327,3 +334,79 @@ func getKubernetesResources(kogitoRes *kogitores.KogitoAppResources) []utilsres.
 
 	return k8sRes
 }
+
+// Ensure that all Kogito images are available before the build.
+func (r *ReconcileKogitoApp) ensureKogitoImageStream(instance *v1alpha1.KogitoApp) (reconcile.Result, error) {
+	isISPresentOnOpenshiftNamespace := true
+
+	imageStreamTag := kogitores.ImageStreamTag
+	if instance.Spec.Build.ImageS2I.ImageStreamTag != "" {
+		imageStreamTag = instance.Spec.Build.ImageS2I.ImageStreamTag
+		log.Infof("Using ImageS2I Tag from CR: %s", imageStreamTag)
+	}
+
+	// First look on the current namespace, if the imagestreams are available, return
+	// and don't check for anything else.
+	for _, key := range kogitores.ImageStreamNameList {
+		is, _ := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: key, Namespace: instance.Namespace}, imageStreamTag)
+		if is == nil {
+			r.isImagestreamReady = false
+		}
+	}
+
+	// look for the Kogito image stream on the default namespace (openshift)
+	// All images must be present, if not create it on current namespace
+	if !r.isImagestreamReady {
+		for _, key := range kogitores.ImageStreamNameList {
+			isOcpNs, _ := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: key, Namespace: kogitores.ImageStreamNamespace}, imageStreamTag)
+			if isOcpNs == nil {
+				isISPresentOnOpenshiftNamespace = false
```

Comment:
Have you tested this condition in an actual cluster? The operator user has the permissions to fetch `is` from `openshift` namespace?

Reply:
Tested on my local cluster running with operator-sdk.
worth a test using olm as well.

Reply:
Should work, but since now that we have our own env, do you mind testing there?

Reply:
Yes, we have the required permissions: https://github.com/kiegroup/kogito-cloud-operator/blob/master/deploy/role.yaml#L11

```yaml
- apiGroups:
      - ""
      - app.kiegroup.org
      - apps.openshift.io
      - image.openshift.io
      - build.openshift.io
      - rbac.authorization.k8s.io
      - route.openshift.io
    resources:
      - "*"
    verbs:
      - "*"
```